server/webhook.go

```diff
@@ -32,7 +34,7 @@ type webhook struct {
 	text          string
 	fields        []*model.SlackAttachmentField
 	notifications []webhookNotification
-	fieldInfo     webhookField
+	fieldInfo     []webhookField
```

Comment:
I might have missed it, but can you point to where we need this to be a slice? (I mean: is there a case where we parse multiple fields for one webhook events? At the moment I think we just make multiple webhook structs and put them into `events []*webhook` below)
I'm not against it, I just wonder if we need it.

Reply:
https://github.com/mattermost/mattermost-plugin-jira/blob/cb7f52f8c3da3c359a3ddd0f8fd44143b398f450/server/subscribe.go#L150
This is where the slice is referenced after the webhook is parsed.

This is for when the webhook has multiple changelog events, and we want to assert that any one of the fields could match the user-defined subscription. At the point of the code linked above runs, the webhook is already parsed, and the context of each change log field has been lossed. A structure needs to persist to hold the context in tact.

Thank you for pointing this out, I had left out the line in `mergeWebhookEvents()` to actually preserve each field's info. I have the test correctly written for checking this functionality now. How do you suggest preserving the information on which fields were changed? We need to have the `fieldId` of the changelog field.

Reply:
This has been removed after using a `Set` for the event type data structure :+1: